app/questionnaire/placeholder_renderer.py

```diff
@@ -66,6 +69,16 @@ def render_placeholder(
         placeholder_data: MutableMapping[str, Any],
         list_item_id: Optional[str],
     ) -> str:
+
+        for placeholder in placeholder_data.get("placeholders", []):
+            if transforms := placeholder.get("transforms"):
+                for transform in transforms:
+                    if check_transform := self.check_transform(transform):
+                        self._block_on_routing_path = check_transform
+                        break
+            if self._block_on_routing_path:
+                break
+
```

Comment:
We can probably extract this to its own method to keep `render_placeholder` segmented:
i.e
```python
    def _check_needs_block_on_routing_path(self, placeholder_data):
        for placeholder in placeholder_data.get("placeholders", []):
            if transforms := placeholder.get("transforms"):
                for transform in transforms:
                    if self.check_transform(transform):
                        return True

        return False
```

And in `render_placeholder()`:
```python
        self._needs_block_on_routing_path = self._needs_block_on_routing_path(
            placeholder_data
        )
```

Reply:
Extracted to own method with suggested solution. Changed the names of the functions.